File: 100.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}
var flag = true
var preOrder = function(p, q){
    if (!flag) {
        return
    }
    if (p == null && q == null) {
        return 
    }
    if (p == null || q == null) {
        flag = false
        return 
    }
    if (p.val != q.val) {
        flag = false
        return 
    }
    preOrder(p.left, q.left)
    preOrder(p.right, q.right)
}
var isSameTree = function(p, q) {
    preOrder(p,q)
    return flag
};

File: 100.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def preOrder(self, p, q):
        if not self.flag:
            return
        if not p and not q:
            return
        if not p or not q:
            self.flag = False
            return
        if p.val != q.val:
            self.flag = False
            return
        self.preOrder(p.left, q.left)
        self.preOrder(p.right, q.right)

    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        self.flag = True
        self.preOrder(p, q)
        return self.flag
